src/agentscope/formatter/_zhipu_formatter.py

```diff
@@ -0,0 +1,348 @@
+from ..message import (
+    Msg,
+    TextBlock,
+    ImageBlock,
+    ToolUseBlock,
+    ToolResultBlock,
+    ThinkingBlock,
+    VideoBlock,
+)
+from ..token import TokenCounterBase
+
+
+def _to_zhipu_image_url(url: str) -> str:
+    """Convert an image url to Zhipu AI format. If the given url is a local
+    file, it will be converted to base64 format. Otherwise, it will be
+    returned directly.
+
+    Args:
+        url (`str`):
+            The local or public url of the image.
+    """
+    support_image_extensions = (
+        ".png",
+        ".jpg",
+        ".jpeg",
+        ".gif",
+        ".webp",
+        ".bmp",
+    )
+
+    parsed_url = urlparse(url)
+    lower_url = url.lower()
+
+    if not os.path.exists(url) and parsed_url.scheme != "":
+        if any(lower_url.endswith(_) for _ in support_image_extensions):
+            return url
+
+    elif os.path.exists(url) and os.path.isfile(url):
+        if any(lower_url.endswith(_) for _ in support_image_extensions):
+            with open(url, "rb") as image_file:
+                base64_image = base64.b64encode(image_file.read()).decode(
+                    "utf-8",
+                )
+            # 根据评审意见，修正Base64格式处理
+            # 使用url而不是parsed_url.path来提取扩展名，以处理本地文件路径
+            extension = url.lower().split(".")[-1]
+            mime_type = f"image/{extension}"
+            return f"data:{mime_type};base64,{base64_image}"
+
+    raise TypeError(f'"{url}" should end with {support_image_extensions}.')
+
+
+class ZhipuChatFormatter(TruncatedFormatterBase):
+    """The class used to format message objects into the Zhipu AI API required
+    format."""
+
+    support_tools_api: bool = True
+    """Whether support tools API"""
+
+    support_multiagent: bool = True
+    """Whether support multi-agent conversation"""
+
+    support_vision: bool = True
+    """Whether support vision models"""
+
+    supported_blocks: list[type] = [
+        TextBlock,
+        ImageBlock,
+        ToolUseBlock,
+        ToolResultBlock,
+        ThinkingBlock,
+        VideoBlock,
+    ]
+    """Supported message blocks for Zhipu AI API"""
+
+    async def _format(
+        self,
+        msgs: list[Msg],
+    ) -> list[dict[str, Any]]:
+        """Format message objects into Zhipu AI API required format.
+
+        Args:
+            msgs (`list[Msg]`):
+                The list of Msg objects to format.
+
+        Returns:
+            `list[dict[str, Any]]`:
+                A list of dictionaries, where each dictionary has "role"
+                and "content" keys.
+        """
+        self.assert_list_of_msgs(msgs)
+
+        messages: list[dict] = []
+        for msg in msgs:
+            content_blocks = []
+            tool_calls = []
+
+            for block in msg.get_content_blocks():
+                typ = block.get("type")
+                if typ == "text":
+                    content_blocks.append({**block})
+
+                elif typ == "thinking":
```

Comment:
This `elif` branch can be eliminated since its condition will inevitably be handled by the fallback `else` clause at line 190, making the code cleaner and more maintainable.